modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationHolder.java

```diff
@@ -0,0 +1,322 @@
+        basicConfig.put(Environment.NUXEO_DATA_DIR, runtimeResolver.apply(Environment.DEFAULT_DATA_DIR));
+        basicConfig.put(Environment.NUXEO_LOG_DIR, homeResolver.apply(Environment.DEFAULT_LOG_DIR));
+        basicConfig.put(Environment.NUXEO_MP_DIR, homeResolver.apply(Environment.DEFAULT_MP_DIR));
+        basicConfig.put(Environment.NUXEO_PID_DIR, homeResolver.apply(Environment.DEFAULT_LOG_DIR));
+        basicConfig.put(Environment.NUXEO_TMP_DIR, homeResolver.apply(Environment.DEFAULT_TMP_DIR));
+        defaultConfig = new Properties(basicConfig);
+        userConfig = new CryptoProperties(defaultConfig);
+        templates = new ArrayList<>();
+    }
+
+    public Path getHomePath() {
+        return home;
+    }
+
+    public Path getNuxeoConfPath() {
+        return nuxeoConf;
+    }
+
+    public Path getTemplatesPath() {
+        return home.resolve(TEMPLATES);
+    }
+
+    /**
+     * Returns the Home of NuxeoRuntime (same as {@code Framework.getRuntime().getHome()}).
+     */
+    public Path getRuntimeHomePath() {
+        return home.resolve(NXSERVER);
+    }
+
+    public Path getConfigurationPath() {
+        return getPropertyAsPath(Environment.NUXEO_CONFIG_DIR);
+    }
+
+    public Path getDataPath() {
+        return getPropertyAsPath(Environment.NUXEO_DATA_DIR);
+    }
+
+    public Path getDumpedConfigurationPath() {
+        return getConfigurationPath().resolve(ConfigurationConstants.FILE_CONFIGURATION_PROPERTIES);
+    }
+
+    public Path getLogPath() {
+        return getPropertyAsPath(Environment.NUXEO_LOG_DIR);
+    }
+
+    public Path getLogConfigPath() {
+        return getHomePath().resolve(LOG4J2_CONF);
+    }
+
+    protected Path getPackagesPath() {
+        return getPropertyAsPath(Environment.NUXEO_MP_DIR);
+    }
+
+    public Path getPidDirPath() {
+        return getPropertyAsPath(Environment.NUXEO_PID_DIR);
+    }
+
+    protected Path getTmpPath() {
+        return getPropertyAsPath(Environment.NUXEO_TMP_DIR);
+    }
+
+    public boolean isLoaded() {
+        return !userConfig.isEmpty();
+    }
+
+    /**
+     * Returns the property value associated with the given {@code key} from the default configuration.
+     */
+    public String getDefaultProperty(String key) {
+        return defaultConfig.getProperty(key);
+    }
+
+    /**
+     * Returns the property value associated with the given {@code key} from the user configuration.
+     * <p>
+     * The method falls back on the default configuration if the user configuration doesn't contain a property value
+     * associated with the given {@code key}.
+     */
+    public String getProperty(String key) {
+        return userConfig.getProperty(key);
+    }
+
+    /**
+     * Returns the property value associated with the given {@code key} from the user configuration.
+     * <p>
+     * The method falls back on the default configuration if the user configuration doesn't contain a property value
```

Comment:
on the given {@code defaultValue} no?

Reply:
after falling back on default configuration, will add a comment.